packages/vite/src/node/config.ts

```diff
@@ -1180,7 +1180,10 @@ async function loadConfigFromBundledFile(
       .toString(16)
       .slice(2)}`
     const fileNameTmp = `${fileBase}.mjs`
-    const fileUrl = `${pathToFileURL(fileBase)}.mjs`
+    // Bun will cache file/directories, and since we just created the file it might throw a ResolveError
+    // Adding a query string with the `?` will bust the cache.
+    // See https://github.com/oven-sh/bun/issues/3216
+    const fileUrl = `${pathToFileURL(fileBase)}.mjs?`
```

Comment:
Not oppose to this change, but I feel like Bun should ultimately fix this too. Currently the issue is closed and appending `?` is _the_ way in Bun?

Reply:
Agreed, we could move forward with this one but it will be good to remove this hack as soon as Bun works out this bug. I think we should only append `?` when `process.versions.bun`.

Reply:
Bun's builtin TypeScript, JSX, ESM, and CommonJS support means transpiling a configuration file ahead of time should be unnecessary. Reading, transpiling, saving and then reading the configuration file again makes Vite start slower than it could. Rather than conditionally adding a `?`, what if we plan to disable this step entirely in Bun?

Reply:
That would lose slight parity with the current config loader, as it tracks the dependencies of the config to trigger a server restart. Personally, I think this "feature" has blocked us from simplifying a lot of things that I'd rather not have it (around Vite 5? Not sure yet).

So I guess we have two options:
1. Use `?` cache bust, but Bun should fix this eventually.
2. Load config with Bun without deps restart trigger. Bun users have slightly different experience.

Reply:
I lean toward using `?` here if we can't get the same experience.

Reply:
@Jarred-Sumner replying to your comment (https://github.com/vitejs/vite/pull/13901#issuecomment-1643899731), I think it would be best if we can implement the least changes on our end here to support Bun. If no2 requires using more Bun-specific APIs, that would likely create another code path to maintain.

Is https://github.com/oven-sh/bun/issues/3216 (the `?` issue) not fixable on Bun's end? I feel like this quirk could affect more libraries in general than only Vite.

Reply:
The reason we put the `?` is because the fix for bun isn't trivial. I agree we should fix this in bun, but it might not happen today for 0.7.